packages/assets-controllers/src/NftDetectionController.ts

```diff
@@ -394,10 +395,16 @@ export class NftDetectionController extends BaseController<
           last_sale && { lastSale: last_sale },
         );
 
-        await this.addNft(address, token_id, nftMetadata, {
-          userAddress: selectedAddress,
-          chainId,
-        });
+        await this.addNft(
```

Comment:
Are there existing tests covering this? If so do they need to be extended to verify that these changes produce the desired effect?

Reply:
so I have [a PR](https://github.com/MetaMask/core/pull/1415) pointed at this one to test atleast part of the additions here.

You're right we should test this a bit more explicitly though.

Reply:
Ah gotcha, okay. I'll take a look at that one.

Reply:
more tests added here: https://github.com/MetaMask/core/pull/1417/commits/7b54bcf9c58a11872e0fa0be5a969f39a00050aa

Reply:
Ah. That's good - but are there any tests that need to be added for NftDetectionController specifically?

Reply:
I've added one here: https://github.com/MetaMask/core/pull/1417/commits/4e9b1ee5d815f12003625bf1f312a6fa263affcb#diff-c48a7b91e7a594a25f0868505f6dd588bbfc783890f0ee2c381399daeba3e04eR316

NftDetection behavior should be unchanged but we should have had this test before.

Reply:
Ah I see. Cool. Thank you!